File: rest/voice/generate-twiml-gather/twiml-gather.5.x.rb
require 'rubygems'
require 'sinatra'
require 'twilio-ruby'

voice_gather = '/voice/handle-gather'

get '/voice' do
  Twilio::TwiML::Response.new do |r|
    r.Say "Hello. It's me.", voice: 'alice', language: 'en-GB'
    r.Play 'https://deved-sample-assets-2691.twil.io/ahoyhoy.mp3'
    r.Gather numDigits: '1', action: voice_gather, method: 'get' do |g|
      g.Say 'To speak to a real person, press 1.'
      g.Say 'Press 2 to record a message for a Twilio educator.'
      g.Say 'Press any other key to start over.'
    end
  end.text
end

File: rest/voice/generate-twiml-gather/twiml-gather.5.x.cs
// In Package Manager, run:
// Install-Package Twilio.Mvc -DependencyVersion HighestMinor

using System.Web.Mvc;
using Twilio.TwiML;

public class VoiceController : Controller
{
    // /Voice
    public ActionResult Index()
    {
        var gather = new Gather (numDigits: 1, action: "/Voice/HandleGather");
        gather.Say("To speak to a real person, press 1.\n" +
            "Press 2 to record a message for a Twilio educator.\n" +
            "Press any other key to start over.");

        var response = new VoiceResponse();
        response.Say("Hello. It's me.", voice: "alice", language: "en-GB");
        response.Play("https://deved-sample-assets-2691.twil.io/ahoyhoy.mp3");
        response.Gather (gather);

        return Content(response.ToString(), "text/xml");
    }
}
